src/net/ossl.cc

```diff
@@ -701,11 +707,17 @@ class session : public enable_shared_from_this<session>, public session_impl {
           return ssl;
       }())
       , _type(t) {
-        bio_ptr in_bio(BIO_new(BIO_s_mem()));
-        bio_ptr out_bio(BIO_new(BIO_s_mem()));
+        bio_ptr in_bio(BIO_new(get_method()));
```

Comment:
If `get_method` throws (on the first attempt), there will probably be a memory leak in openssl.

Reply:
within the OpenSSL library or in this implementation?

Reply:
> within the OpenSSL library or in this implementation?

In OpenSSL, `bio_new` isn't exception-safe.

Reply:
Ignore this, I misread how that works.